pkg/devfile/parser/configurables.go

```diff
@@ -0,0 +1,218 @@
+			return d.setPorts(arrValue...)
+		case "memory":
+			return d.setMemory(strValue)
+		}
+
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+
+}
+
+func (d DevfileObj) DeleteConfiguration(parameter string) error {
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.setMetadataName("")
+		case "ports":
+			return d.removePorts()
+		case "memory":
+			return d.setMemory("")
+		}
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a configurable parameter in the devfile", parameter)
+}
+
+func (d DevfileObj) IsSet(parameter string) bool {
+
+	if parameter, ok := AsDevfileSupportedParameter(parameter); ok {
+		switch parameter {
+		case "name":
+			return d.getMetadataName() != ""
+		case "ports":
+			return d.hasPorts()
+		case "memory":
+			return d.getMemory() != ""
+		}
+	}
+	return false
+
+}
+
+func (d DevfileObj) setMetadataName(name string) error {
+	metadata := d.Data.GetMetadata()
+	d.Data.SetMetadata(name, metadata.Version)
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) AddEnvVars(otherList config.EnvVarList) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
+			component.Container.Env = currentlist.Merge(otherList).ToDevfileEnv()
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) RemoveEnvVars(keys []string) error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+
+			currentlist := config.NewEnvVarListFromDevfileEnv(component.Container.Env)
+			envList, err := config.RemoveEnvVarsFromList(currentlist, keys)
+			if err != nil {
+				return err
+			}
+			component.Container.Env = envList.ToDevfileEnv()
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) setPorts(ports ...string) error {
+	components := d.Data.GetComponents()
+	endpoints, err := portsToEndpoints(ports...)
+	if err != nil {
+		return err
+	}
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Endpoints = endpoints
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) removePorts() error {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if component.Container != nil {
+			component.Container.Endpoints = nil
+			d.Data.UpdateComponent(component)
+		}
+	}
+	return d.WriteYamlDevfile()
+}
+
+func (d DevfileObj) hasPorts() bool {
+	components := d.Data.GetComponents()
+	for _, component := range components {
+		if len(component.Container.Endpoints) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (d DevfileObj) setMemory(memory string) error {
+	components := d.Data.GetComponents()
```

Comment:
We should validate memory limit, not specifically here may be during `odo push`, if memory limit specified in wrong format like `756g`, `odo config set` `odo push` `k8s api`silently ignores that, and component gets pushed successfully without applying limits.

Reply:
opened an issue https://github.com/openshift/odo/issues/3791 for this